hbs-minifier-plugin.js

```diff
@@ -88,15 +98,26 @@ class HBSMinifierPlugin {
 
     return { name: 'hbs-minifier-plugin', visitor };
   }
-
-  transform(ast) {
-    let plugin = HBSMinifierPlugin.createASTPlugin();
-
-    this.syntax.traverse(ast, plugin.visitor);
-
-    return ast;
-  }
-
 }
 
-module.exports = HBSMinifierPlugin;
+module.exports = function(config) {
+  config = config || {};
+  config = assignDefaultValues(config);
+
+  return class HBSMinifierPlugin extends BasePlugin {
+
+    transform(ast) {
+      let startLoc = ast.loc ? ast.loc.start : {};
+      /*
+        checking for line and column to avoid registering the plugin for ProgramNode inside a BlockStatement.
+      */
+      if (startLoc.line !== 1 || startLoc.column !== 0) {
+        return ast;
+      }
```

Comment:
I don't understand why this is necessary. Isn't the `transform()` function only called for top-level elements? The AST traversal should be handled by the `traverse()` call below.

Reply:
In ember 2.15, the transform is called for all the `Program` Nodes. I am not sure why.

Reply:
that is a bug in Ember 2.15 that is already fixed again afaik

/cc @rwjblue

Reply:
Do you want this issue to be patched up or should I keep it as an issue for that particular Ember versions?

Reply:
I think we should no guard against that. The transform that we do should be idempotent so the only advantage of this guard should be less work to do for the buggy versions.